src/main/java/tech/jhipster/lite/module/infrastructure/secondary/javadependency/gradle/GradleCommandHandler.java

```diff
@@ -429,18 +430,31 @@ public void handle(AddGradlePlugin command) {
     Assert.notNull(COMMAND, command);
 
     switch (command.plugin()) {
-      case GradleCorePlugin plugin -> declarePlugin(plugin.id().get());
+      case GradleCorePlugin plugin -> declarePlugin(plugin.id().get(), command.buildProfile());
       case GradleCommunityPlugin plugin -> {
-        declarePlugin("alias(libs.plugins.%s)".formatted(applyVersionCatalogReferenceConvention(pluginAlias(plugin))));
+        declarePlugin(
+          "alias(libs.plugins.%s)".formatted(applyVersionCatalogReferenceConvention(pluginAlias(plugin))),
+          command.buildProfile()
+        );
         versionsCatalog.addPlugin(plugin);
       }
+      case GradleProfilePlugin plugin -> {
+        declarePlugin(
+          """
+          id("%s")\
+          """.formatted(plugin.dependency().get().id().groupId()),
+          command.buildProfile()
+        );
+        versionsCatalog.addLibrary(plugin.dependency().get());
+        addDependencyToBuildGradle(plugin.dependency().get(), projectFolder.filePath(PLUGIN_BUILD_GRADLE_FILE), false);
+      }
```

Comment:
@murdos : should I delete `GradlePluginDependency` record and use `JavaDependency` directly at `GradleProfilePlugin`?

The code below will be cleaner:
```suggestion
      case GradleProfilePlugin plugin -> {
        declarePlugin(
          """
          id("%s")\
          """.formatted(plugin.dependency().id().groupId()),
          command.buildProfile()
        );
        versionsCatalog.addLibrary(plugin.dependency());
        addDependencyToBuildGradle(plugin.dependency(), projectFolder.filePath(PLUGIN_BUILD_GRADLE_FILE), false);
      }
```

Reply:
IMO JavaDependency has too much information (scope, classifier, optional, exclusions,...), so maybe `GradlePluginDependency` should only contains:
```java
private final GroupId groupId;
private final ArtifactId artifactId;
private final Optional<VersionSlug> versionSlug;
```
Maybe, internally, GradleCommandHandler could convert a GradlePluginDependency to a JavaDependency, for factorization and internal API reuse, but at least the public API is clean.
WDYT?

Reply:
I agree 👍👍👍